docs/resource.json

```diff
@@ -0,0 +1,20 @@
+{
+  "ID": "resource",
+  "Properties": {
+    "Resources": {
+      "resource_a": {
+        "Plugin": "instance-file",
+        "Type": "a",
+        "Properties": {
+          "Note": "Depends on [[ .resource_b ]]"
```

Comment:
What if instead of using a different delimiter `[[`, we still use the same `{{}}` but introduce a new function instead?   

So we'd have something that looks like:

```
"Properties" : {
  "Note" : "Depends on {{ resource \"some/path/or/key\" }}",
}
```
With the `pkg/template` package it's easy to do, and you can bind different function implementations to the function name `resource` so that maybe in one stage of template processing, you are binding the keyword `resource` to a function that can construct a graph by adding a node whenever `resource` function is called.  And on another pass, you are returning actual, computed values from somewhere.

The escaping `"` is unfortunate, but if we switch to YAML in the future, it won't be a big eyesore.  I think this would at least make things look consistent with templates used elsewhere in the project.

Will this work?